gateway/ln-gateway/src/ln.rs

```diff
@@ -25,3 +27,43 @@ pub struct LnRpcRef {
 
 #[derive(Debug)]
 pub struct LightningError(pub Option<i32>);
+
+/// The core-lightning `htlc_accepted` event's `amount` field has a "msat" suffix
+/// TODO: Can we standardize this for all LN implementations?
+fn as_fedimint_amount<'de, D>(amount: D) -> Result<Amount, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let amount = String::deserialize(amount)?;
+    Ok(Amount::from_msat(
+        amount[0..amount.len() - 4].parse::<u64>().unwrap(),
+    ))
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Htlc {
+    #[serde(deserialize_with = "as_fedimint_amount")]
+    pub amount: Amount,
+    pub cltv_expiry: u32,
+    pub cltv_expiry_relative: u32,
+    pub payment_hash: bitcoin_hashes::sha256::Hash,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Onion {
+    pub payload: String,
+    #[serde(rename = "type")]
+    pub type_: String,
+    pub short_channel_id: String,
+    #[serde(deserialize_with = "as_fedimint_amount")]
+    pub forward_amount: Amount,
+    pub outgoing_cltv_value: u32,
+    pub shared_secret: bitcoin_hashes::sha256::Hash,
+    pub next_onion: String,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct HtlcAccepted {
+    pub htlc: Htlc,
+    pub onion: Onion,
+}
```

Comment:
These structs are specific to CLN so they should probably stay in `cln.rs`

Reply:
Ack. I'm hoping LND helps us define generic versions of these so the signature of `ReceiveInvoice` more or less stays the same

Reply:
I expect we'll have 2 sets of structs that are different. These are CLN / LND implementation details.

Once we implement LND we can make our own struct that just contains the information which the gateway cares about. We could implement `Into` for between the CLN / LND specific structs into the version that the gateway handles.